### ISSUE TITLE
InvalidDatasourceError: Datasource URL should use prisma:// protocol

### ISSUE THREAD

**Issue by jacobhq** [prisma/prisma#11526]
I'm using prisma in nextjs, and the env urls start with `postgres://`. I'm working in a gitpod container. When I run `next dev` **or** build for production and run `next start`, I receive this error:

```
InvalidDatasourceError: Datasource URL should use prisma:// protocol
```

If I change my urls to start with `prisma://`, I get:

```
InvalidDatasourceError: No valid API key found in the datasource URL
```

Any ideas? Help would be much appreciated! I've listed my schema and versions of prisma below.

```
datasource db {
  provider             = "postgresql"
  url                  = env("DATABASE_URL")
  shadowDatabaseUrl    = env("DATABASE_MIGRATE_URL")
  referentialIntegrity = "prisma"
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["dataproxy", "referentialIntegrity"]
}

model User {
  id           Int            @id @default(autoincrement())
  WishlistItem WishlistItem[]
}

model Product {
  id           Int            @id @default(autoincrement())
  name         String
  price        Int
  salePrice    Int            @default(0)
  imageUrl     String
  flag         String?
  description  String?
  buyUrl       String
  marketplace  String?
  WishlistItem WishlistItem[]
}

model WishlistItem {
  id        Int     @id @default(autoincrement())
  userId    Int
  productId Int
  Product   Product @relation(fields: [productId], references: [id])
  User      User    @relation(fields: [userId], references: [id])
}

```

```json
    "@prisma/client": "^3.8.1",
    "prisma": "3.8.1",
    "next": "12.0.8",
```

```js
export const getServerSideProps = async ({}) => {
  const prisma = new PrismaClient()
  const products = await prisma.product.findMany()
  return { props: { products } }
}
```

**Comment by janpio** [comment]
How are you building your Prisma Client? From the preview features it seems you might be trying to use the Data Proxy, correct?

**Comment by jacobhq** [comment]
Thanks for pointing that out! This is not a bug it's just the fact that I wasn't using the dataproxy. Thanks!

**Comment by jacobhq** [comment]
> **TLDR**: I solved this by removing `dataproxy` from the `previewFeatures` part of my schema.

@Zineeddine998 Prisma has a feature called Data Proxy, a tool for db connection management in serverless environments (better explained in this [blog post from prisma](https://www.prisma.io/blog/prisma-data-proxy-xb16ba0p21)), which I was not using for this project, but had enabled it as I copied code from a previous project that was. As you can see below, it was ledt enabled.

> `previewFeatures = ["dataproxy", "referentialIntegrity"]`

When using Data Proxy, you are given a proxy url starting with the `prisma://` protocol. The url in my env var started with the `postgres://` protocol, and were therefore not accepted. I was unable to deduce this from the error message, and opened the issue, and @janpio's comment helped me realise. I then solved this by removing `dataproxy` from the `previewFeatures` part of my schema.

**Comment by pradok** [comment]
I am using Prisma Data Proxy but I am getting this issue `InvalidDatasourceError: Datasource URL should use prisma:// protocol` in my integration tests using docker. Is there a way to fix this for tests so it doesn't use proxy?

**Comment by jacobhq** [comment]
> I am using Prisma Data Proxy but getting this issue InvalidDatasourceError: Datasource URL should use prisma:// protocol in my integration tests using docker. Is there a way to fix this for tests so it doesn't use proxy?

@pradok So you are using the dataproxy everywhere but in tests? Please could you give me more info?

**Comment by pradok** [comment]
> > I am using Prisma Data Proxy but getting this issue InvalidDatasourceError: Datasource URL should use prisma:// protocol in my integration tests using docker. Is there a way to fix this for tests so it doesn't use proxy?
> 
> @pradok So you are using the dataproxy everywhere but in tests? Please could you give me more info?

@jacobhq  I have got CI/CD pipeline running tests in a docker container so I guess Proxy doesn't make sense(?). But do I need to use Proxy in tests? I think it would be mainly for production environment. 

I am using AWS Aurora with Postgres and running AWS serverless.

**Comment by BrennerSpear** [comment]
I am all of a sudden getting this same error. I didn't change anything. I committed unrelated code, ran `yarn dev` again, and now I get this error.

I'm using `pscale connect wallet-activity-local dev --port 3309` so yes, I am using the dataProxy

I've tried
* switching to `prisma://` which gives me the "no API key" error
* removed `dataProxy`
* both at the same time, just to cover my bases, even if it doesn't make sense

What am I missing?

**Comment by janpio** [comment]
> I'm using `pscale connect wallet-activity-local dev --port 3309` so yes, I am using the dataProxy

This does not actually look like a Prisma Data Proxy, but a PlanetScale local proxy. For that you can use a normal `mysql://...` connection uri.

**Comment by BrennerSpear** [comment]
@janpio sorry - you're right. I just went and read through all of the dataProxy stuff.

I just set up data platform, and now things are working, but what's weird was it was working for most of today, and then it randomly stopped, and I had to go set this up. Did something change on the prisma side that made me have to change it?

**Comment by janpio** [comment]
No - Prisma's Data Proxy is purely optional. This error message can really only happen if you configure the generation process of Prisma Client to use the data proxy preview feature and engine type.

**Comment by Danubius1st** [comment]
I work on a Nextjs app with:
next@13.4.9
graphql@16.7.1
graphql-yoga@4.0.3
@apollo/client@3.7.17
@apollo/server@4.7.5
@prisma/client@5.0.0
prisma@5.0.0

I get these errors on vercel logs:
`404 [GET] /_vercel/insights/view`
`405 [POST] /api/graphql InvalidDatasourceError: Datasource URL must use prisma:// protocol when --accelerate or --data-proxy are used`

I ran out of solutions. Any hint would be appreciated.

**Comment by janpio** [comment]
Open a new issue please @Danubius1st and provide all the information the template asks for. Thank you.

**Comment by Danubius1st** [comment]
https://github.com/prisma/prisma/issues/20237
